api/api/src/main/scala/uk/ac/wellcome/platform/api/services/ElasticsearchService.scala

```diff
@@ -69,6 +75,38 @@ class ElasticsearchService(elasticClient: ElasticClient)(
         }
     }
 
+  def executeMultiSearchRequest(request: MultiSearchRequest)
+    : Future[Either[ElasticError, List[SearchResponse]]] =
+    spanFuture(
+      name = "ElasticSearch#executeQuery",
+      spanType = "request",
+      subType = "elastic",
+      action = "query"
+    ) {
+      debug(s"Sending ES multirequest: ${request.show}")
+      val transaction = Tracing.currentTransaction
+      withActiveTrace(elasticClient.execute(request))
+        .map(toEither)
+        .map { response =>
+          response.right.flatMap {
+            case MultiSearchResponse(items) =>
+              val results = items.map(_.response)
+              val error = results.collectFirst { case Left(err) => err }
+              error match {
+                case Some(err) => Left(err)
+                case None =>
+                  Right(
+                    results.collect {
+                      case Right(resp) =>
+                        transaction.addLabel("elasticTook", resp.took)
```

Comment:
It looks like this might not be doing what you think it is - it's repeatedly updating the value of `elasticTook`, once for each item in `results`. As you say, probably best to either sum them or - maybe better - sum them _and_ have them as separate labels so we can see how long the constituent queries took.

Reply:
Ah ok thanks for clarifying what each field is used for.

I think I will just update it to sum the values: as this is something which isn't hanging around for long I don't reckon we need more granularity than that.

Reply:
> sum them and have them as separate labels so we can see how long the constituent queries took.

This please